whylog/tests/tests_teacher/test_rule_validation.py

```diff
@@ -1,80 +1,122 @@
 from whylog.config.filename_matchers import WildCardFilenameMatcher
 from whylog.config.log_type import LogType
-from whylog.tests.tests_teacher import TestBase
+from whylog.constraints import TimeConstraint
+from whylog.tests.tests_teacher import TestRuleBase
 
 from whylog.teacher.rule_validation_problems import (  # isort:skip
     NoEffectParserProblem, NotSetLogTypeProblem, NotUniqueParserNameProblem, ParserCountProblem
 )  # yapf: disable
+from whylog.constraints.validation_problems import (  # isort:skip
+    MinGreaterThatMaxProblem, ParamConversionProblem
+)
 
 
-class TestValidationBase(TestBase):
-    def _in_rule_problems(self, problem):
+class TestValidationBase(TestRuleBase):
+    def _check_if_parser_has_problem(self, parser_id, problem):
+        validation_result = self.teacher.validate()
+        problems = validation_result.parser_problems.get(parser_id)
+        if problems is None:
+            return False
+        return problem in problems
+
+    def _check_if_rule_has_problem(self, problem):
         validation_result = self.teacher.validate()
         return problem in validation_result.rule_problems
 
-    def _in_parser_problems(self, parser_id, problem):
+    def check_if_constraint_has_problem(self, constraint_id, problem):
         validation_result = self.teacher.validate()
-        problems = validation_result.parser_problems.get(parser_id)
+        problems = validation_result.constraint_problems.get(constraint_id)
         if problems is None:
             return False
         return problem in problems
 
     def _initial_validation_check(self):
         validation_result = self.teacher.validate()
+        print(validation_result.rule_problems)
```

Comment:
I think this function is never used, because CI passes and this is a python3-style print, which requires `from __future__ import print_function`, which is missing